packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -52,6 +73,10 @@ const awaitVisitor = {
 };
 
 export default function(path: NodePath, file: Object, helpers: Object) {
+  path.traverse(yieldVisitor, {
+    visited: new WeakSet(),
+  });
+
   path.traverse(awaitVisitor, {
```

Comment:
Can you merge `yieldVisitor` and `awaitVisitor`, to avoid traversing the function body twice?

Reply:
initially i went with combined visitors, but unfortunately it had issue - but unfortunately [ForOfStatement visitor](https://github.com/babel/babel/blob/d7fd8888fa838c6c7ed07820a558ca77ee31dd0b/packages/babel-helper-remap-async-to-generator/src/index.js#L24) in the `awaitVisitor` produces new `yield` statements between [those lines](https://github.com/babel/babel/blob/d7fd8888fa838c6c7ed07820a558ca77ee31dd0b/packages/babel-helper-remap-async-to-generator/src/for-await.js#L13-L26). I couldn't figure out later how to distinguish them without introducing some brittle code, because for the `yieldVisitor` those statements need to get skipped